dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpServiceContextImpl.java

```diff
@@ -2892,8 +2903,16 @@ final protected VirtualConnection sendFullOutgoing(WsByteBuffer[] wsbb, HttpBase
             if (this instanceof HttpInboundServiceContextImpl) {
                 hisc = (HttpInboundServiceContextImpl) this;
             }
-            if (hisc != null && !(hisc.getLink() instanceof H2HttpInboundLinkWrap)) {
-                createEndOfBodyChunk();
+            if (hisc != null) {
```

Comment:
Originally this was checking that the link wasn't an H2HttpInboundLinkWrap, and not doing it now would seem to be a change in behavior. Additionally, by not checking the instanceof of the hisc, this conditional could potentially do a hisc.getSupress0ByteChunk() on an object that isn't an HttpInboundServiceContextImpl, which could lead to exceptions during the runtime. I recommend only doing the new code when the hisc is proven to be HttpInboundServiceContextImpl.